File: prog3/trab_java/utilitarios/capes/producoes/Partitura.java
package utilitarios.capes.producoes;

public class Partitura extends Producao implements Comparable<Partitura> {
    String editora;
    String formacao_instrumental;

    public Partitura(String natureza, String idioma, String cidade, int qtdPags, String editora, String formacao_instrumental){
        super(null, natureza, idioma, cidade, qtdPags);
        this.editora = editora;
        this.formacao_instrumental = formacao_instrumental;
        this.tipo = 5;
    }

    @Override
    public int compareTo(Partitura o) {
        return this.natureza.compareTo(o.natureza) != 0 ?
        this.natureza.compareTo(o.natureza) : this.editora.compareTo(o.editora) != 0 ?
        this.editora.compareTo(o.editora) : this.cidade.compareTo(o.cidade) != 0 ?
        this.cidade.compareTo(o.cidade) : this.formacao_instrumental.compareTo(o.formacao_instrumental) != 0 ?
        this.formacao_instrumental.compareTo(o.formacao_instrumental) : this.qtdPags - o.qtdPags;
    }

    public static void imprimeCabecalho(){
        System.out.println("Natureza;Editora;Cidade;Formacao;Paginas");
    }  

    @Override
    public String toString(){
        return (this.natureza + ";" + this.editora + ";" + this.cidade + ";" + this.formacao_instrumental + ";" + this.qtdPags_string);
    } 
}

File: prog3/trab_cpp/Partitura.cpp
#include "Partitura.hpp"

Partitura::Partitura(string natureza, string cidade, int qtd_pags, string editora, string formacao): Producao("", natureza, "", cidade, qtd_pags, 28){
    this->editora = editora;
    this->formacao = formacao;
}

Partitura::~Partitura(){}

/* Metodo que retorna a string que contem as informacoes de uma producao em formato csv. No caso, reorna uma string no seguinte formato:
 "<Natureza>;<Editora>;<Cidade>;<Formacao>;<Paginas>"*/
string Partitura::csv(){
    stringstream stream;
    stream << natureza << ";" << editora << ";" << cidade << ";" << formacao << ";" << _int_string(qtd_pags);
    return stream.str();
}

/* Metodo que retorna a string cabecalho. */
string Partitura::cabecalho(){
    return "Natureza;Editora;Cidade;Formacao;Paginas";
}

/* Operador de comparacao de uma producao com outra de mesmo subtipo. */
bool Partitura::operator < (Producao& o){
    Partitura& prod = (Partitura&)o;
    int cmp = natureza.compare(prod.natureza);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = editora.compare(prod.editora);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = cidade.compare(prod.cidade);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = formacao.compare(prod.formacao);
    if(cmp<0) return true; else if(cmp>0) return false;
    cmp = qtd_pags - prod.qtd_pags;
    if(cmp<0) return true; else return false;
}